### PR TITLE
Implement 'Configure Project Environment' command

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: src/ProjectEnvironmentConfiger.ts [hunk 1]
```diff
@@ -23,22 +23,31 @@ const ioTContainerizedProjectModule =
...
 
-    let openInNewWindow = false;
-    // If current window contains other project, open the created project in new
-    // window.
-    if (vscode.workspace.workspaceFolders &&
-        vscode.workspace.workspaceFolders.length > 0) {
-      openInNewWindow = true;
+    if (!vscode.workspace.workspaceFolders) {
+      const message =
+          'No folder is currently open in Visual Studio Code. Please select a folder first.';
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: You have not yet opened a folder in Visual Studio Code. Please select a folder first.

File: src/ProjectEnvironmentConfiger.ts [hunk 2]
```diff
@@ -23,22 +23,31 @@ const ioTContainerizedProjectModule =
...
+      vscode.window.showWarningMessage(message);
+      return;
+    }
+
+    const rootPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: >.workspaceFolders[0] [](start = 37, length = 20)

Please check the length of the array (vscode.workspace.workspaceFolders.length) before you access it with index.

File: src/ProjectEnvironmentConfiger.ts [hunk 3]
```diff
@@ -91,28 +108,143 @@ export class ProjectEnvironmentConfiger {
...
   }
 
-  private async configureWorkspaceProjectEnv() {
-    // TODO
+  /**
+   * Configure Arduino project: Add tasks.json file
+   * Ask to overwrite if file already exists.
+   * @returns true - configuration success. false - configuration cancel.
+   */
+  private async configureProjectEnv(
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: Seems this function is to configure project environment for all platform type, NOT just for 'Arduino'.
If so, please correct the comment.

File: src/ProjectEnvironmentConfiger.ts [hunk 4]
```diff
@@ -91,28 +108,143 @@ export class ProjectEnvironmentConfiger {
...
+
+    return;
   }
 
-  private async configureWorkspaceProjectEnv() {
-    // TODO
+  /**
+   * Configure Arduino project: Add tasks.json file
+   * Ask to overwrite if file already exists.
+   * @returns true - configuration success. false - configuration cancel.
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: >false - configuration cancel [](start = 44, length = 28)

Would there be an error except for a 'Cancel' operation?

File: src/ProjectEnvironmentConfiger.ts [hunk 5]
```diff
@@ -91,28 +108,143 @@ export class ProjectEnvironmentConfiger {
...
+  private async askToOverwriteFile(fileName: string):
+      Promise<boolean|undefined> {
+    const overwriteTasksJsonOption: vscode.QuickPickItem[] = [];
+    overwriteTasksJsonOption.push(
+        {label: `Yes`, description: ''}, {label: `No`, description: ''});
+
+    const overwriteSelection =
+        await vscode.window.showQuickPick(overwriteTasksJsonOption, {
+          ignoreFocusOut: true,
+          placeHolder: `${fileName} already exists. Overwrite?`
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: >Overwrite [](start = 52, length = 9)

Do you want to overwrite?

File: src/ProjectEnvironmentConfiger.ts [hunk 6]
```diff
@@ -65,22 +74,30 @@ export class ProjectEnvironmentConfiger {
...
+          if (!templateJson) {
+            throw new Error('Fail to load template list.');
+          }
+
           if (platform === PlatformType.Arduino) {
-            await this.configureWorkspaceProjectEnv();
+            const arduinoTaskName = 'Arduino Task';
+            const res = await this.configureProjectEnv(
+                context, channel, telemetryContext, scaffoldType, templateJson,
+                rootPath, arduinoTaskName);
```
Comment (lirenhe) [inline src/ProjectEnvironmentConfiger.ts]: Is arduinoTaskName the 'templatename'?

File: src/ProjectEnvironmentConfiger.ts [hunk 7]
```diff
@@ -91,28 +108,143 @@ export class ProjectEnvironmentConfiger {
...
+   * selection.
+   */
+  private async askToOverwriteFile(fileName: string):
+      Promise<boolean|undefined> {
+    const overwriteTasksJsonOption: vscode.QuickPickItem[] = [];
+    overwriteTasksJsonOption.push(
+        {label: `Yes`, description: ''}, {label: `No`, description: ''});
+
+    const overwriteSelection =
+        await vscode.window.showQuickPick(overwriteTasksJsonOption, {
```
Comment (lirenhe) [inline src/ProjectEnvironmentConfiger.ts]: is it a good practice to use QuickPick to select 'yes' or 'no'?

File: src/ProjectEnvironmentConfiger.ts (hunk 4 above)
Comment (dilin-MS) [reply src/ProjectEnvironmentConfiger.ts]: Fixed.

File: src/ProjectEnvironmentConfiger.ts [hunk 8]
```diff
@@ -22,25 +23,35 @@ const ioTContainerizedProjectModule =
     impor('./Models/IoTContainerizedProject') as
     typeof import('./Models/IoTContainerizedProject');
 
+enum overWriteLabel {
```
Comment (dooriya) [inline src/ProjectEnvironmentConfiger.ts]: >overWrite [](start = 5, length = 9)

lower case 'overwrite'

Comment (dilin-MS) [reply src/ProjectEnvironmentConfiger.ts]: Fixed.

**Pull request merged**